### PR TITLE
Refactoring tests

### PR DESCRIPTION
**Pull request by grzanka**
During work on #95 I've figured out that the test suite can be improved.
The score of this PR does a refactor of the tests directory.
The tests are grouped according to the simulator.

Screenshot 2023-05-13 at 22 26 27

### CODE REVIEW

File: tests/fluka/test_fluka_converter.py [hunk 1]
```diff
@@ -63,26 +62,19 @@
...
-@pytest.fixture
-def output_dir(tmp_path_factory) -> str:
-    """Fixture that creates a temporary dir for testing converter output."""
-    output_dir = tmp_path_factory.mktemp(_Test_dir)
-    return output_dir
 
 def test_parser(parser) -> None:
     """Check if parser is created correctly."""
     assert parser.info["version"] == "unknown"
     assert parser.info["simulator"] == "fluka"
```
Comment (hendzeld) [inline tests/fluka/test_fluka_converter.py:74]: There is no check for `label` value.

```suggestion
    assert parser.info["simulator"] == "fluka"
    assert parser.info["label"] == ""
```

File: tests/shieldhit/test_shieldhit_converter.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+from converter.api import run_parser
+from converter.shieldhit.parser import Parser
+
+
+def test_parser(sh12a_parser : Parser) -> None:
+    """Check if parser is created correctly"""
+    assert sh12a_parser
+    assert sh12a_parser.info['version'] == ''
+    assert sh12a_parser.info['simulator'] == 'shieldhit'
```
Comment (hendzeld) [inline tests/shieldhit/test_shieldhit_converter.py:15]: There is no check for `label` value.

```suggestion
    assert sh12a_parser.info['simulator'] == 'shieldhit'
    assert sh12a_parser.info['label'] == ''
```

File: tests/topas/test_topas_converter.py [hunk 3]
```diff
@@ -1,10 +1,9 @@
-import pytest
 from pathlib import Path
+import pytest
 from converter.api import get_parser_from_str, run_parser
 from converter.common import Parser
-import json
 
-_Config_str = """s:Ge/MyBox/Type     = "TsBox"
+_excpected_topas_input_content = """s:Ge/MyBox/Type     = "TsBox"
```
Comment (Bogdan2423) [inline tests/topas/test_topas_converter.py:6]: typo:
```suggestion
_expected_topas_input_content = """s:Ge/MyBox/Type     = "TsBox"
```

File: tests/topas/test_topas_converter.py [hunk 4]
```diff
@@ -66,30 +65,24 @@
...
-def test_if_config_created(output_dir) -> None:
+def test_if_config_created(topas_parser: Parser, project_topas_json: dict, tmp_path: Path) -> None:
     """Check if topas_config.txt file created"""
-    output_file = output_dir.joinpath('topas_config.txt')
+    output_file = tmp_path / 'topas_config.txt'
+    run_parser(topas_parser, project_topas_json, tmp_path)
     with output_file.open(mode='r') as f:
-        assert f.read() == _Config_str
\ No newline at end of file
+        assert f.read() == _excpected_topas_input_content
```
Comment (Bogdan2423) [inline tests/topas/test_topas_converter.py:88]:
```suggestion
        assert f.read() == _expected_topas_input_content
```

File: tests/fluka/test_fluka_converter.py [hunk 5]
```diff
@@ -1,9 +1,8 @@
 import pytest
-from os import path
-from converter.fluka.parser import FlukaParser
+from converter.common import Parser
 from converter.api import get_parser_from_str, run_parser
 
-_Config_content = """TITLE
+_excpected_fluka_input_content = """TITLE
```
Comment (Bogdan2423) [inline tests/fluka/test_fluka_converter.py:5]:
```suggestion
_expected_fluka_input_content = """TITLE
```

File: tests/fluka/test_fluka_converter.py [hunk 6]
```diff
@@ -63,26 +62,20 @@
...
 
-def test_if_inp_created(parser, output_dir) -> None:
+def test_if_inp_created(parser, tmp_path) -> None:
     """Check if fl_sim.inp file created."""
-    run_parser(parser, {}, output_dir)
-    with open(path.join(output_dir, "fl_sim.inp")) as f:
-        assert f.read() == _Config_content
+    run_parser(parser, {}, tmp_path)
+    with open(tmp_path / "fl_sim.inp") as f:
+        assert f.read() == _excpected_fluka_input_content
```
Comment (Bogdan2423) [inline tests/fluka/test_fluka_converter.py:81]:
```suggestion
        assert f.read() == _expected_fluka_input_content
```

**Review by Bogdan2423 — APPROVED** [review]
lgtm

**Review by hendzeld — APPROVED** [review]
LGTM

**Pull request merged**